src/Streamly/Internal/Data/Stream/Async/Channel/Operations.hs

```diff
@@ -281,6 +278,9 @@ _fromChannelD svar = D.Stream step FromSVarInit
     step _ (FromSVarLoop sv (ev : es)) = do
         case ev of
             ChildYield a -> return $ D.Yield a (FromSVarLoop sv es)
+            ChildStopChannel -> do
```

Comment:
Does this mean that this function was initially incorrect? It did not account for the exception `ChildStop`?

Reply:
Yes, should have handled the ChannelStop exception in the previous change.